core/src/main/java/org/apache/james/mime4j/stream/MimeEntity.java

```diff
@@ -395,10 +398,13 @@ private void advanceToBoundary() throws IOException {
         }
     }
 
+    @SuppressWarnings( "resource" )
     private EntityStateMachine nextMessage() {
         // optimize nesting of streams returning the "lower" stream instead of
         // always return dataStream (that would add a LineReaderInputStreamAdaptor in the chain)
         InputStream instream = currentMimePartStream != null ? currentMimePartStream : inbuffer;
+        // instream is not closed here, so supress the warning.
+        // TODO: is that really intended??
```

Comment:
Todos never get fixed. The comment is also unhelpful. If it’s correct we should enforce the behaviour in a test. If it’s broken we should raise a ticket.